src/common/fullstory.ts

```diff
@@ -32,8 +37,43 @@ function initStorage() {
     return;
   }
   const enableRecording = Math.random() < FULLSTORY_RECORD_PERCENT;
-  localStorage.setItem(
-    FULLSTORY_RECORD_KEY,
-    enableRecording ? RecordSession.TRUE : RecordSession.FALSE,
-  );
+  try {
+    localStorage.setItem(
+      FULLSTORY_RECORD_KEY,
+      enableRecording ? RecordSession.TRUE : RecordSession.FALSE,
+    );
+  } catch (err) {
+    // It doesn't matter if we can't write to localStorage, we only need this
+    // for 1% of the users.
+    console.warn(`Error writing to localStorage`, err);
+  }
+}
+
+// From https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API/Using_the_Web_Storage_API#testing_for_availability
+function storageAvailable(storageType: string) {
+  let storage;
+  const x = '__storage_test__';
+  try {
+    // @ts-ignore
+    storage = window[storageType];
+    storage.setItem(x, x);
+    storage.removeItem(x);
+    return true;
+  } catch (e) {
+    return (
+      e instanceof DOMException &&
+      // everything except Firefox
+      (e.code === 22 ||
+        // Firefox
+        e.code === 1014 ||
+        // test name field too, because code might not be present
+        // everything except Firefox
+        e.name === 'QuotaExceededError' ||
+        // Firefox
+        e.name === 'NS_ERROR_DOM_QUOTA_REACHED') &&
+      // acknowledge QuotaExceededError only if there's something already stored
+      storage &&
+      storage.length !== 0
```

Comment:
FWIW- We could probably treat "quota exceeded" as if storage isn't available and just replace this whole catch block with `return false;`.  But probably (hopefully) won't come up in practice.